### PR TITLE
Add mock tests for the dynamic client

### LINKED ISSUE #952: [tests] Add mock test client for dynamic client
Add mock test client for dynamic client to streamline unit testing.
We should be able to pass the errors to expect while executing via dynamic client.
Ref implementation for mock client: https://github.com/knative/client/blob/master/pkg/sources/v1alpha2/apiserver_client_mock.go#L26

### PR DESCRIPTION
**Pull request by priyshar01**
## Description

Adding mock test client for the dynamic client.

## Changes
Added mock tests for Dynamic client.

## Reference

Fixes #952

### CODE REVIEW

**Comment by priyshar01** [comment]
@googlebot I fixed it

File: pkg/dynamic/client_mock_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	"knative.dev/client/pkg/util/mock"
+)
+
+func TestMockKnDynamicClient(t *testing.T) {
+
+	client := NewMockKnDyanmicClient(t)
+
+	recorder := client.Recorder()
+
+	// Record all services
```
Comment (navidshaikh) [inline pkg/dynamic/client_mock_test.go:30]:
```suggestion

```

File: pkg/dynamic/client_mock_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	recorder.ListCRDs(mock.Any(), nil, nil)
+	recorder.ListSourcesTypes(nil, nil)
+	recorder.ListSources(mock.Any(), nil, nil)
+
+	// Call all service
```
Comment (navidshaikh) [inline pkg/dynamic/client_mock_test.go:35]:
```suggestion

```

File: pkg/dynamic/client_mock.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+// ListCRDs returns list of installed CRDs in the cluster and filters based on the given options
+func (c *MockKnDynamicClient) ListCRDs(options metav1.ListOptions) (*unstructured.UnstructuredList, error) {
+	call := c.recorder.r.VerifyCall("ListCRDs", options)
+	return call.Result[0].(*unstructured.UnstructuredList), mock.ErrorOrNil(call.Result[1])
+}
+
+// ListSourcesTypes returns installed knative eventing sources CRDs
+func (dr *ClientRecorder) ListSourcesTypes(ulist *unstructured.UnstructuredList, err error) {
+	dr.r.Add("ListSourcesTypes", []interface{}{}, []interface{}{ulist, err})
+
```
Comment (navidshaikh) [inline pkg/dynamic/client_mock.go:78]:
```suggestion
```

Comment (navidshaikh) [reply pkg/dynamic/client_mock.go:78]: also remove empty lines in following methods

File: pkg/dynamic/client_mock_test.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	client.ListCRDs(metav1.ListOptions{})
+	client.ListSourcesTypes()
+	client.ListSources(WithTypeFilter("blub"))
+	// Validate
```
Comment (navidshaikh) [inline pkg/dynamic/client_mock_test.go:39]: missing RawClient test

**Comment by priyshar01** [comment]
Will raise a fresh PR for this.

**Pull request closed without merging**